src/main/java/com/example/newslettercore/domain/newsletter/model/Template.java

```diff
@@ -0,0 +1,74 @@
+package com.example.newslettercore.domain.newsletter.model;
+
+import com.example.newslettercore.domain.exception.TemplateCanalsNotFoundException;
+import com.example.newslettercore.domain.exception.TemplateTextInvalidException;
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.logging.log4j.util.Strings;
+
+import java.util.List;
+
+@Getter
+public class Template {
+
+    private String id;
+
+    private List<String> channels;
+
+    private String text;
+
+    @Setter
+    private Newsletter newsletter;
+
+    public Template(String id, List<String> channels, String text, Newsletter newsletter) {
+
+        this(channels, text, newsletter);
+        this.id = id;
+    }
+
+    public Template(List<String> channels, String text, Newsletter newsletter) {
+
+        validateTemplateData(channels, text);
+
+        this.channels = channels;
+        this.text = text;
+        this.newsletter = newsletter;
+    }
+
+    private void validateTemplateData(List<String> canals, String text) {
+
+        validateCanals(canals);
+        validateText(text);
+    }
+
+    private void validateCanals(List<String> canals) {
+
+        if (CollectionUtils.isEmpty(canals)) {
+            throw new TemplateCanalsNotFoundException();
+        }
+    }
+
+    private void validateText(String text) {
+
+        if (Strings.isBlank(text)) {
+            throw new TemplateTextInvalidException();
+        }
+    }
+
+    public Template updateTemplate(List<String> canals, String text) {
+
+        if (!CollectionUtils.isEmpty(canals)) {
+            this.channels = canals;
+        }
+
+        if (Strings.isNotEmpty(text)) {
+            this.text = text;
+        }
+
+        newsletter.getTemplates().removeIf(template -> template.getId().equals(id));
```

Comment:
Po co usuwamy i dodajemy ten sam template ? Dodatkowo, template, jak na razie, nie musi wiedzieć do jakiego newslettera nalerzy, więć to pole może być usunięte.

Reply:
Wczesniej umawialismy sie, ze zapisuje do bazy caly newsletter, a nie template. Zeby to zrobic musze zaaktualizowac odpowiedni template w newsleterze

Reply:
A ten obiekt template na którym wywołujemy updateTemplate, nie będzie tym samym obiektem (tą samą instancją) który będzie znajdować się w liscie templates w Newsletter ?

Reply:
Nie bedzie, sprawdzalem to